### PR TITLE
Many upgrades to cookiecutter

### LINKED ISSUE #7: Add default email/logging config
Logging and email config should be backported from CCXCon/TeachersPortal to this cookiecutter.

### LINKED ISSUE #37: Add API example which integrates with redux and has functioning tests

### LINKED ISSUE #40: Javascript code coverage
Seems to be an oversight, we should configure our test environment to output code coverage for javascript, maybe using Istanbul: http://stackoverflow.com/questions/16633246/code-coverage-with-mocha

### LINKED ISSUE #44: Add ipdb to test_requirements.txt

### LINKED ISSUE #47: Create shared webpack config
https://github.com/mitodl/micromasters/pull/211/

### LINKED ISSUE #48: Fix onUpdate
https://github.com/mitodl/micromasters/pull/235

### LINKED ISSUE #49: Update OSX setup
https://github.com/mitodl/micromasters/pull/356#issuecomment-220682275
https://github.com/mitodl/micromasters/pull/400

### LINKED ISSUE #52: Add favicon
https://github.com/mitodl/micromasters/pull/359

### LINKED ISSUE #55: Add fake API function
This would include api.js, api_test.js, fetch_mock

### LINKED ISSUE #57: Upgrade to Python 3.5
See https://github.com/mitodl/micromasters/pull/466
and https://github.com/mitodl/micromasters/pull/499

### LINKED ISSUE #58: Add flow

### LINKED ISSUE #60: Update uwsgi
This should include changes to `uwsgi.ini` and the change we made to have docker use that instead of `runserver`

### LINKED ISSUE #61: Add nginx configuration

### LINKED ISSUE #62: Add requirements.in

### LINKED ISSUE #63: Add celery

### PR DESCRIPTION
**Pull request by rhysyngsun**
Fixes #63 
Fixes #62 
Fixes #61 
Fixes #60 
Fixes #58 
Fixes #57 
Fixes #55 
Fixes #52 
Fixes #49 
Fixes #48 
Fixes #47 
Fixes #44 
Fixes #40 
Fixes #37 
Fixes #7

### CODE REVIEW

**Comment by rhysyngsun** [comment]
We have a requirements.in, but no requirements.txt. In order to create requirements.txt we need to run `pip-compile`, which requires building the container, which requires a valid requirements.txt.

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py [hunk 1]
```diff
@@ -181,22 +227,39 @@ def get_var(name, default):
 EMAIL_SUPPORT = get_var('{{ cookiecutter.project_name|upper }}_SUPPORT_EMAIL', 'support@example.com')
 DEFAULT_FROM_EMAIL = get_var('{{ cookiecutter.project_name|upper }}_FROM_EMAIL', 'webmaster@localhost')
 
+MAILGUN_URL = get_var('MAILGUN_URL', 'https://api.mailgun.net/v3/micromasters.mit.edu')
```
Comment (rhysyngsun) [inline {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py]: Micromasters ref.

Comment (noisecapella) [inline {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py]: I left that in on purpose, despite its name we use it in bootcamp as well. @pdpinch just to confirm, it's fine to use it here as well?

**Comment by noisecapella** [comment]
I think that's all the changes now, I'll update the body to link the issues

**Comment by rhysyngsun** [comment]
:+1:

**Pull request merged**